### PR TITLE
Rename ChainerIO to PFIO

### LINKED ISSUE #118: Rename to `pfio`
As the `Chainer` has entered the maintenance phrase, we are planing on renaming this library to `pfio`.
But to reduce the porting effort in users code, we can keep the `chainerio` name and let users to import from both name

### PR DESCRIPTION
**Pull request by belldandyxtq**
We rename the ChainerIO to PFIO as `Chainer` has entered the
maintenance phase.

An alias of `chainerio` has created to keep the backward compatibility.
This closes #118

### CODE REVIEW

File: pfio/cache/file_cache.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ class FileCache(cache.Cache):
     '''Cache system with local filesystem
 
     Stores cache data in local temporary files, created in
-    ``~/.chainer/chainerio/cache`` by default. Cache data is
+    ``~/.chainer/pfio/cache`` by default. Cache data is
```
Comment (belldandyxtq) [inline pfio/cache/file_cache.py:83]: Maybe it is better to put under `.pfn/`?

File: examples/memnn/babi_chainerio.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import collections
 # CHAINERIO import
```
Comment (shu65) [inline examples/memnn/babi_chainerio.py:2]: CHAINERIO -> PFIO?

File: examples/memnn/babi_chainerio.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def read_data(vocab, path):
     data = []
     all_data = []
     # CHAINERIO add
```
Comment (shu65) [inline examples/memnn/babi_chainerio.py:78]: CHAINERIO -> PFIO?

File: pfio/cache/file_cache.py (hunk 1 above)
Comment (shu65) [reply pfio/cache/file_cache.py:83]: How about `~/.pfio/cache` ?

**Pull request merged**